reference/zrc6.scilla

```diff
@@ -0,0 +1,963 @@
+transition Burn(token_id: Uint256)
+  RequireNotPaused;
+  (* Check if token exists *)
+  opt_token_owner <- token_owners[token_id];
+  match opt_token_owner with
+  | None =>
+    error = TokenNotFoundError;
+    Throw error
+  | Some token_owner =>
+    RequireOwnerOrOperator token_owner;
+    (* Destroy existing token *)
+    delete token_owners[token_id];
+    delete spenders[token_id];
+
+    (* subtract one from the balance *)
+    UpdateBalance sub_operation token_owner;
+    (* subtract one from the total supply *)
+    current_supply <- total_supply;
+    new_supply = builtin sub current_supply one;
+    total_supply := new_supply;
+    e = {
+      _eventname: "Burn";
+      burn_address: token_owner;
+      token_id: token_id
+    };
+    event e;
+    msg_to_sender = { 
+      _tag: "ZRC6_BurnCallback";
+      _recipient: _sender;
+      _amount: Uint128 0;
+      burn_address: token_owner;
+      token_id: token_id
+    };
+    msgs = one_msg msg_to_sender;
+    send msgs
+  end
+end
+
+(* Adds `to` as minter. *)
+(* @Requirements: *)
+(* - `_sender` should be the contract owner. Otherwise, it should throw `NotContractOwnerError` *)
+(* - `to` should not be already a minter. Otherwise, it should throw `MinterFoundError` *)
+transition AddMinter(to: ByStr20)
+  RequireContractOwner;
+  has_minter <- exists minters[to];
+  match has_minter with
+  | True => 
+    error = MinterFoundError;
+    Throw error
+  | False =>
+    (* Add minter *)
+    minters[to] := true
+  end;
+  e = { 
+    _eventname: "AddMinter";
+    to: to
+  };
+  event e;
+  msg_to_sender = { 
+    _tag: "ZRC6_AddMinterCallback";
+    _recipient: _sender;
+    _amount: Uint128 0;
+    to: to
+  };
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* Removes `to` from minter. *)
+(* @Requirements: *)
+(* - `_sender` should be the contract owner. Otherwise, it should throw `NotContractOwnerError` *)
+(* - `to` should be already a minter. Otherwise, it should throw `MinterNotFoundError` *)
+transition RemoveMinter(to: ByStr20)
+  RequireContractOwner;
+  has_minter <- exists minters[to];
+  match has_minter with
+  | True => 
+    delete minters[to]
+  | False =>
+    error = MinterNotFoundError;
+    Throw error
+  end;
+  e = { 
+    _eventname: "RemoveMinter";
+    to: to
+  };
+  event e;
+  msg_to_sender = { 
+    _tag: "ZRC6_RemoveMinterCallback";
+    _recipient: _sender;
+    _amount: Uint128 0;
+    to: to
+  };
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* Adds `to` as spender of `token_id`. *)
+(* There can only be one spender per token at any given time. *)
+(* @Requirements: *)
+(* - `token_id` should exist. Otherwise, it should throw `TokenNotFoundError` *)
+(* - `_sender` should be a token owner or an operator. Otherwise, it should throw `NotOwnerOrOperatorError` *)
+(* - `to` should not be already a spender. Otherwise, it should throw `SpenderFoundError` *)
+transition AddSpender(to: ByStr20, token_id: Uint256)
```

Comment:
can consider to change `to` to `new_spender`

Reply:
Let's go with `spender` as we discussed.